### PR TITLE
Added share icons

### LINKED ISSUE #66: Connect the share stage button
image

### PR DESCRIPTION
**Pull request by UsamaSaleem14**
image

image

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -5,39 +5,59 @@
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <meta name="theme-color" content="#000000" />
+    <meta name="title" content="DaoSeeder" data-react-helmet="true" />
     <meta
       name="description"
-      content="Decentralized crowdfunding with trust."
+      content="World's first decentralized Crowdfunding platform."
```
Comment (TBomberman) [inline public/index.html]: call it Trusted Decentralized Crowdfunding Platform

File: package.json [hunk 2]
```diff
@@ -18,10 +18,12 @@
     "react": "^18.2.0",
     "react-datepicker": "^4.10.0",
     "react-dom": "^18.2.0",
+    "react-helmet": "^6.1.0",
```
Comment (TBomberman) [inline package.json]: Please explain how helmet works next time.

File: src/components/HelmetMetaData/index.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import { Helmet } from "react-helmet";
+import { useLocation } from "react-router-dom";
+import { IReactHelmet } from "../../interfaces/IReactHelmet";
+
+export default function HelmetMetaData(props: IReactHelmet) {
+  const location = useLocation();
+  const currentUrl = window.location.hostname + location.pathname;
+  const title = props.title
+    ? props.title
+    : "Daoseeder: World's First Decentralized crowdfunding platform";
```
Comment (TBomberman) [inline src/components/HelmetMetaData/index.tsx]: Change to the same text mentioned above.

File: src/components/Stages/index.tsx [hunk 4]
```diff
@@ -99,6 +108,13 @@ function Stage() {
 
   return (
     <>
+      {stageData?.stage ? (
+        <HelmetMetaData
+          title={stageData.stage.name}
+          image={""}
```
Comment (TBomberman) [inline src/components/Stages/index.tsx]: why are you having blank image and discription?

Comment (UsamaSaleem14) [reply src/components/Stages/index.tsx]: Removed this as we are now copying the url

File: src/components/HelmetMetaData/index.tsx (hunk 3 above)
Comment (UsamaSaleem14) [reply src/components/HelmetMetaData/index.tsx]: Removed this as we are now copying the url

File: package.json (hunk 2 above)
Comment (UsamaSaleem14) [reply package.json]: Removed this as we are now copying the url

File: public/index.html (hunk 1 above)
Comment (UsamaSaleem14) [reply public/index.html]: Done

File: package.json [hunk 5]
```diff
@@ -22,6 +22,7 @@
     "react-icons": "^4.7.1",
     "react-router-dom": "^6.6.2",
     "react-scripts": "5.0.1",
+    "react-share": "^4.4.1",
```
Comment (TBomberman) [inline package.json]: I don't see this being used anywhere. Can this be removed?

File: public/index.html [hunk 6]
```diff
@@ -5,39 +5,39 @@
     <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <meta name="theme-color" content="#000000" />
+    <meta name="title" content="DaoSeeder" />
     <meta
       name="description"
-      content="Decentralized crowdfunding with trust."
+      content="World's first Trusted Decentralized Crowdfunding Platform."
```
Comment (TBomberman) [inline public/index.html]: Remove World's first.

File: src/hooks/useSingleCampaignHandler.ts [hunk 7]
```diff
@@ -139,5 +139,18 @@ export const useSingleCampaignHandler = () => {
     }
   };
 
-  return { campaign, mediaLinkIdx, prevItem, nextItem, campaigns, allStages };
+  const copyLink = () => {
+    window.navigator.clipboard.writeText(window.location.href);
```
Comment (TBomberman) [inline src/hooks/useSingleCampaignHandler.ts:143]: Does this get the whole link as is?

File: src/interfaces/IReactHelmet.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+export interface IReactHelmet {
```
Comment (TBomberman) [inline src/interfaces/IReactHelmet.ts]: can this file be removed?

File: yarn.lock [hunk 9]
```diff
@@ -3368,6 +3368,13 @@
   dependencies:
     "@types/react" "*"
 
+"@types/react-helmet@^6.1.6":
```
Comment (TBomberman) [inline yarn.lock]: Can you update yarn lock after removing packages from packages.json?

File: src/hooks/useSingleCampaignHandler.ts (hunk 7 above)
Comment (UsamaSaleem14) [reply src/hooks/useSingleCampaignHandler.ts:143]: Yes

**Review by TBomberman — APPROVED** [review]

**Review by TBomberman — CHANGES_REQUESTED** [review]
Why is yarn.lock need to be changed now that there are no changes to package.json? Can the yarn.lock be reverted?

**Comment by TBomberman** [comment]
Can you also merge main to here to resolve conflicts?

**Pull request merged**